src/app/courses-page/course-search/course-search.component.spec.ts

```diff
@@ -1,24 +1,40 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
 
 import { CourseSearchComponent } from './course-search.component';
 
 describe('CourseSearchComponent', () => {
   let component: CourseSearchComponent;
   let fixture: ComponentFixture<CourseSearchComponent>;
-
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      declarations: [CourseSearchComponent]
-    }).compileComponents();
-  }));
+  let buttonEl: DebugElement;
+  const defaultValue = 'course1';
 
   beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [CourseSearchComponent],
+      imports: [FormsModule]
+    });
     fixture = TestBed.createComponent(CourseSearchComponent);
     component = fixture.componentInstance;
+    buttonEl = fixture.debugElement.query(By.css('.search-btn'));
     fixture.detectChanges();
   });
 
+  afterEach(() => {
+    fixture.destroy();
```

Comment:
![image](https://user-images.githubusercontent.com/46322232/99000975-c221db00-2542-11eb-9a43-09b8793187c1.png)
Maybe I misunderstood this comment.
But I deleted afterEach and such error doesn't appear.

Reply:
that's because you don't recompile the whole module for each test, but rather compile just a single component.
Maybe this will make it a little faster, but you should be clearing all mutated data on the mock services if you have some